ipu-plugin/pkg/ipuplugin/lifecycleservice.go

```diff
@@ -230,7 +254,249 @@ func configureChannel(mode, daemonHostIp, daemonIpuIp string) error {
+	if err != nil {
+		return fmt.Errorf("failed to create remote file.sh: %s", err)
+	}
+	defer initFile.Close()
+
+	initScript := `#!/bin/sh
+CURDIR=$(pwd)
+WORKDIR=$(dirname $(realpath $0))
+cd $WORKDIR	
+	if [ -e load_custom_pkg.sh ]; then
+	# Fix up the cp_init.cfg file
+	./load_custom_pkg.sh
+	fi
+	python /usr/bin/scripts/cfg_acc_apf_x2.py
+fi
+cd $CURDIR
+`
+	_, err = initFile.Write([]byte(initScript))
+	if err != nil {
+		return fmt.Errorf("failed to write to file.sh: %s", err)
+	}
+
+	err = initFile.Sync()
+	if err != nil {
+		return fmt.Errorf("failed to sync file.sh: %s", err)
+	}
+
+	// Start a session.
+	session, err := client.NewSession()
+	if err != nil {
+		return fmt.Errorf("failed to create session: %s", err)
+	}
+	defer session.Close()
+
+	cmd := exec.Command("sh", "-c", "cat /proc/sys/kernel/random/uuid | sha256sum | head -c 2")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err = cmd.Run()
+	if err != nil {
+		return fmt.Errorf("command execution error: %s", err)
+	}
+	hashedChars := out.String()
+
+	macAddress := fmt.Sprintf("00:00:00:0a:%s:15", hashedChars)
+	fmt.Println("Updated MAC address:", macAddress)
+
+	shellScript := fmt.Sprintf(`#!/bin/sh
+	CP_INIT_CFG=/etc/dpcp/cfg/cp_init.cfg
+	echo "Checking for custom package..."
+	if [ -e linux_networking.pkg ]; then
+		echo "Custom package linux_networking.pkg found. Overriding default package"
+		cp linux_networking.pkg /etc/dpcp/package/
+		rm -rf /etc/dpcp/package/default_pkg.pkg
+		ln -s /etc/dpcp/package/linux_networking.pkg /etc/dpcp/package/default_pkg.pkg
+		sed -i 's/sem_num_pages = 1;/sem_num_pages = 25;/g' $CP_INIT_CFG
+		sed -i 's/pf_mac_address = "00:00:00:00:03:14";/pf_mac_address = "%s";/g' $CP_INIT_CFG
+		sed -i 's/acc_apf = 4;/acc_apf = 19;/g' $CP_INIT_CFG
+		sed -i 's/comm_vports = ((\[5,0\],\[4,0\]));/comm_vports = ((\[5,0\],\[4,0\]),(\[0,3\],\[4,2\]));/g' $CP_INIT_CFG
+	else
+		echo "No custom package found. Continuing with default package"
+	fi
+	`, macAddress) // Insert the MAC address variable into the script.
+
+	commands := fmt.Sprintf(`
+	cat <<'EOF' > /work/scripts/load_custom_pkg.sh
+	%s
+	EOF
+	`, shellScript)
+
+	uuidFilePath := "/work/uuid"
+	uuidFile, err := sftpClient.Create(uuidFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to create remote uuid file: %s", err)
+	}
+	defer uuidFile.Close()
+
+	// Write the new MAC address to the uuid file.
+	_, err = uuidFile.Write([]byte(macAddress + "\n"))
+	if err != nil {
+		return fmt.Errorf("failed to write to uuid file: %s", err)
+	}
+
+	// Ensure that the uuid file is written to the remote filesystem.
+	err = uuidFile.Sync()
+	if err != nil {
+		return fmt.Errorf("failed to sync uuid file: %s", err)
+	}
+
+	// Run a command on the remote server and capture the output.
+	var stdoutBuf bytes.Buffer
+	session.Stdout = &stdoutBuf
+	err = session.Run(commands)
+	if err != nil {
+		return fmt.Errorf("failed to run commands: %s", err)
+	}
+
+	session, err = client.NewSession()
```

Comment:
previous session closes once a command is finished, hence a new session.

could try put multiple stuff into a single session call, but this way, we see which specific command fails.
thoughts?